src/v/http/client.cc

```diff
@@ -75,7 +75,9 @@ void client::check() const {
 
 ss::future<client::request_response_t> client::make_request(
   client::request_header&& header, ss::lowres_clock::duration timeout) {
-    vlog(http_log.trace, "client.make_request {}", header);
+    if (http_log.is_enabled(ss::log_level::trace)) {
+        vlog(http_log.trace, "client.make_request {}", header);
+    }
```

Comment:
```
Our
logger formats them even if the actual log level is higher than trace.
This commit adds log level checks before formatting anything.
```

can you describe in the commit/pr why the output string is formatted even if the logger isn't enabled. I don't think that is the intended behavior, and we should probably consider fix it (though perhaps not in this PR).

it looks like the vlog macro builds a call to seastar logger which std::forwards but those references aren't touched if the log level isn't enabled. what am I missing?